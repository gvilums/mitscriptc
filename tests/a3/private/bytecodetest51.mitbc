function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [100000, 1],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [x],
			instructions = 
			[
				load_global	0
				load_const	0
				swap
				gt
				not
				if	6
				load_global	0
				load_const	1
				add
				store_global	0
				goto	-10
				load_global	0
				return
			]
		}
	],
	constants = [0, "y is "],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [x, y, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	2
		load_func	1
		alloc_closure	0
		store_global	3
		load_func	2
		alloc_closure	0
		store_global	4
		load_const	0
		store_global	0
		load_func	3
		alloc_closure	0
		store_global	1
		load_global	2
		load_const	1
		load_global	1
		call	0
		add
		call	1
		pop
	]
}