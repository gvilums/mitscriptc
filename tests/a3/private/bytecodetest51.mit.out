y is 100000
